src/qat/purr/compiler/runtime.py

```diff
@@ -383,6 +384,7 @@ def execute_instructions(
 ):
     if config is None:
         config = CompilerConfig()
+    qatconfig.validate(config)
     config.validate(hardware)
```

Comment:
complexity of validation is building up here, lots of things that look similar but do unclear operations

goes in line with a comment here: https://github.com/oqc-community/qat/pull/271#discussion_r1859257014

Reply:
I agree with the sentiment but I think we need to get the feature to main and then iteratively refactor. Let's land some of the big ticket items (pipelines, hardware_model refactors) then refactor this later.